### PR TITLE
[BD-03] [TNL-7414] [BB-2855] Discussion APIs redux integration etc. 

### PR DESCRIPTION
**Pull request by xitij2000**
This PR adds all the data APIs and redux boilerplate for discussions.

**JIRA tickets**: TNL-7414

**Dependencies**: https://github.com/edx/frontend-app-discussions/pull/1

**Sandbox URL**: NA

**Testing instructions**:

For the purpose of testing the internals of redux have been exposed to allow working via the browser console since this only adds internal APIs. All API methods have been added to window.api, all thunks to window.thunks and the store has been added as well.

For instance you can test the comment adding feature by running the following in your browser console:
``store.dispatch(thunks.addComment("test comment", <thread_id>))``

If you have the comments page open at this time you can see the comment appear in real-time.

**Reviewers**
- [ ] @pcockwell
- [ ] edX reviewer[s] TBD

### CODE REVIEW

File: src/discussions/posts/data/api.js [hunk 1]
```diff
@@ -1,14 +1,35 @@
 /* eslint-disable import/prefer-default-export */
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
-import { API_BASE_URL } from '../../../data/constants';
 
-export async function getCourseThreads(
+ensureConfig([
+  'LMS_BASE_URL',
+], 'Comments API service');
```
Comment (pcockwell) [inline src/discussions/posts/data/api.js:7]: Should this be `Posts API service`?

File: src/discussions/utils.js [hunk 2]
```diff
@@ -10,3 +10,27 @@ export function buildIntlSelectionList(options, intl, messages) {
...
+  Object.keys(params)
+    .forEach(
+      (paramName) => {
+        const paramValue = params[paramName];
+        if (paramValue) {
+          url.searchParams.set(paramName, paramValue);
+        }
+      },
+    );
+}
```
Comment (pcockwell) [inline src/discussions/utils.js:29]: This doesn't appear to be used anywhere - should it be removed?

**Review by pcockwell — CHANGES_REQUESTED** [review]

File: src/discussions/posts/data/thunks.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  };
+}
+
+export function removeThread(threadId) {
+  return async (dispatch) => {
+    try {
+      dispatch(deleteThreadRequest({ threadId }));
+      await deleteThread(threadId);
+      dispatch(postThreadSuccess({ threadId }));
```
Comment (pcockwell) [inline src/discussions/posts/data/thunks.js:128]: This needs to be `deleteThreadSuccess`

**Comment by pcockwell** [comment]
I tested the following `thunks` methods:

```
addComment
editComment
removeComment
createNewThread
updateExistingThread
removeThread
```

All of them worked, with the exception of `removeThread`, which did still remove the thread, but failed to update the UI. I've added a comment where the necessary change for that needs to be. I have not yet tested any fetch methods given that there are no return values, or UI changes from them

**Comment by xitij2000** [comment]
@pcockwell I've pushed an update that addresses your points

**Review by pcockwell — APPROVED** [review]
:+1: Approving dependent on the merging of #1  

- [x] I tested this: I tested these thunks methods:
```
addComment
editComment
removeComment
createNewThread
updateExistingThread
removeThread
```
- [x] I read through the code

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. Checked that the code builds when using Node v14.15.0 (LTS).
2. Checked that the API calls were working. I've tested `api.getThreads` directly and `addComments` through a store dispatch method (then checked that the comment was created in the LMS discussions tab).
- [ ] I read through the code: no, this was already reviewed before.
- [ ] ~~I checked for accessibility issues~~ NA
- [ ] Includes documentation: No.

@xitij2000 Based on the previous approval, this is good to go.
I've checked that the builds work and that the API methods are working fine.

**Pull request merged**